File: protocol/src/test/go/protocol/SimpleObject.go
package protocol

type SimpleObject struct {
	c int
	g bool
}

func (protocol SimpleObject) protocolId() int16 {
	return 104
}

func (protocol SimpleObject) write(buffer *ByteBuffer, packet any) {
	if buffer.WritePacketFlag(packet) {
		return
	}
	var message = packet.(*SimpleObject)
	buffer.WriteInt(message.c)
	buffer.WriteBool(message.g)
}

func (protocol SimpleObject) read(buffer *ByteBuffer) any {
	var packet = new(SimpleObject)
	if !buffer.ReadBool() {
		return packet
	}
	var result0 = buffer.ReadInt()
	packet.c = result0
	var result1 = buffer.ReadBool()
	packet.g = result1
	return packet
}

File: protocol/src/test/python/pyProtocol/SimpleObject.py

class SimpleObject:

    c = 0  # int
    g = False  # bool

    def protocolId(self):
        return 104

    @classmethod
    def write(cls, buffer, packet):
        if buffer.writePacketFlag(packet):
            return
        buffer.writeInt(packet.c)
        buffer.writeBool(packet.g)
        pass

    @classmethod
    def read(cls, buffer):
        if not buffer.readBool():
            return None
        packet = SimpleObject()
        result0 = buffer.readInt()
        packet.c = result0
        result1 = buffer.readBool() 
        packet.g = result1
        return packet

